noir-circuits/blob/src/main.nr

```diff
@@ -186,356 +73,14 @@ fn blob_to_fields__tightly_packed(blob: [F; FIELDS_PER_BLOB]) -> [Field; NOIR_FI
         for j in 1..254 {
             let k = i * 254 + j;
```

Comment:
You likely mean `let k = (i - FIELDS_PER_BLOB) * 254 + j;` here as otherwise you're going to be reading way past the end of `top_bits`.

The reason you're getting extra `array_get`s is that in the case where you're reading past the end of the array we're unable to simplify the instruction despite having a constant index.

Reply:
Whoops, thanks for spotting that!
So Noir doesn't throw an error when you try to read an array out of bounds? I'd have expected a compile-time or run-time error being thrown? We know the size of the array at compile-time, and we know the (incorrectly computed) value of k? (Just wanting to understanding this, because I must be thinking about compilers incorrectly)

Reply:
The tests should throw an error during exexution, if that's not happening then something is wrong.

re compile-time error: we removed most of these as compile time errors for cases such as this result in issues for tests where performing an OOB array read would prevent the test from executing which interferes with debugging.